Docs: Updating UI extensions how-to guides with instructions on adding static information in plugin.json

mckn opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/plugin-tools/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:
Adds missing plugin.json configuration examples to UI extensions documentation. The guides previously mentioned updating metadata but lacked specific examples, causing confusion when extensions didn't work.

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer**:

mckn committed:
docs(ui-extensions): Include plugin.json documentation

mckn committed:
updated.

mckn committed:
reset the correct file.

mckn committed:
wip.

mckn committed:
sip

mckn committed:
wip

mckn committed:
Merge branch 'main' into mckn/added-info-on-static-parts-UI-extensions

mckn committed:
wip

hugohaggmark reviewed (approved):
Awesome this will make a big difference 🙌

sunker commented on docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md:
```diff
@@ -12,7 +12,7 @@ keywords:
 ---
 
-Extensions are links or React components in an app plugin. Extensions are linked to an extension point and they either render in the core Grafana UI or in another app plugin. They can also work as functions returning values. 
+Extensions are links or React components in an app plugin. Extensions are linked to an extension point and they either render in the core Grafana UI or in another app plugin. They can also work as functions returning values.
```
```suggestion
Extensions are links, functions or React components defined in an app plugin. Each extension is associated with an extension point and can render either in the core Grafana UI or within another app plugin.
```

sunker commented on docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md:
```diff
@@ -245,14 +245,11 @@ For more information, see the `plugin.json` [reference](../../reference/metadata
-
-
-
+1. **Check the Grafana version** - link and component extensions are only supported after Grafana version **`>=10.1.0`**. `addLink()` and `addComponent()` are only supported in versions **>=`11.1.0`**.
```
What about functions?

mckn replied:
Great catch, I will have a look and update the docs.

sunker reviewed (commented):
Yes this is a huge improvement! 

Looks like the docs related to the `addFunction` API is missing in `register-an-extension.md`. Can we add a "Work with function extensions" section?

mckn committed:
Merge branch 'main' into mckn/added-info-on-static-parts-UI-extensions

mckn committed:
fix: resolve syntax errors and typos in UI extensions documentation

- Fix missing = in context object definition
- Fix missing closing div tag in JSX example
- Fix typo 'callling' to 'calling'

mckn committed:
updated docs according to the new structure.

mckn committed:
fixed according to suggestion in PR.

mckn committed:
Updated according to feedback.

mckn committed:
Merge branch 'main' into mckn/added-info-on-static-parts-UI-extensions

leventebalogh reviewed (approved):
Very nice updates! 👏
